ovos_core/skill_installer.py

```
@@ -226,17 +226,85 @@ def pip_uninstall(self, packages: list,
+        if "setup.py" in file_names and "pyproject.toml" not in file_names and "setup.cfg" not in file_names:
+            LOG.warning(f"validate_skill: '{owner}/{repo}' uses only setup.py — legacy packaging, rejecting")
+            return False
+
+        # fetch the modern packaging manifest to scan for incompatible base classes
+        for manifest in ("pyproject.toml", "setup.cfg"):
+            if manifest not in file_names:
+                continue
+            manifest_url = f"https://raw.githubusercontent.com/{owner}/{repo}/HEAD/{manifest}"
+            try:
+                manifest_resp = requests.get(manifest_url, timeout=3)
+            except Exception as exc:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)